src/tests/Microsoft.PowerFx.Interpreter.Tests.Shared/IRTests/RewritingIRVisitorTests.cs

```diff
@@ -74,6 +74,43 @@ public void Rewrite100x(string expr, string expectedIR, object expected)
             }
         }
 
+        // This test is to ensure a binary op chain is rewritten and does not get blocked by the depth limit.
+        [Theory]
+        [InlineData("&&", "And")]
+        [InlineData("And", "And")]
+        [InlineData("||", "Or")]
+        [InlineData("Or", "Or")]
+        public void BinaryOpIRRewriteTest(string op, string func)
+        {
+            var sbExpr = new StringBuilder($"true {op} If(true {op} true {op} true, true)");
+            var sbExpected = new StringBuilder($"{func}(True, (If({func}(True, (True), (True)), (True)))");
+
+            for (var i = 0; i < 1000; i++)
```

Comment:
>1000 [](http://example.com/codeflow?start=32&length=4)

Jus for kicks, can we set this limit to 100k and verify that doesn't cause a stack overflow?